db/commands/passenger.js

```diff
@@ -0,0 +1,127 @@
+const db = require( '../config' ).db
+
+class Passenger {
+
+  constructor( id, passengerName, origin, destination, trainNumber, stationName ) {
+    this.id = id
+    this.passengerName = passengerName
+    this.origin = origin
+    this.destination = destination
+    this.trainNumber = trainNumber
+    this.stationName = stationName
+  }
+
+  static getPassengerID( name ) {
+    return db.one(`SELECT * FROM passengers WHERE passenger_name = $1`, name)
+    .then( passenger => {
+      return passenger.id
+    })
+  }
+
+  static getPassengerName( id ) {
+    return db.one(`SELECT * FROM passengers WHERE id = $1`, id)
+    .then( passenger => {
+      return passenger.passenger_name
+    })
+  }
+
+  static getTicket( name ) {
+    return db.one(`SELECT * FROM passengers WHERE passenger_name = $1`, name)
+    .then( passenger => {
+      return {'origin':passenger.origin,'destination':passenger.destination}
+    })
+  }
+
+  setCurrentStation( station ) {
+    this.stationName = station
+    return db.none(`UPDATE passengers SET station_name = $2 WHERE passenger_name = $1`,
+    [ this.passengerName, station ])
+  }
+
+  buyTicket( destination ) {
+    this.destination = destination
+    return db.none(`UPDATE passengers SET destination = $2 WHERE passenger_name = $1`,
+      [ this.passengerName, destination ])
+  }
+
+  useTicket() {
+    return db.one(`SELECT * FROM trains WHERE current_station = $1`, this.stationName)
+    .then( train => {
+      this.trainNumber = train.train_number
+      this.stationName = null
+      this.update( 'trainNumber', this.trainNumber )
```

Comment:
if you dont these could silently fail. Also whatever code is waiting for `useTicket` to finish will fire too soon.